src/name/verify.rs

```diff
@@ -49,45 +49,39 @@ pub fn verify_cert_dns_name_or_ip(
     cert: &crate::EndEntityCert,
     dns_name_or_ip: DnsNameOrIpRef,
 ) -> Result<(), Error> {
-    match dns_name_or_ip {
-        DnsNameOrIpRef::DnsName(dns_name) => verify_cert_dns_name(cert, dns_name),
-        DnsNameOrIpRef::IpAddress(ip_address) => {
-            let ip_address = match ip_address {
-                IpAddressRef::IpV4AddressRef(_, ref ip_address_octets) => {
-                    untrusted::Input::from(ip_address_octets)
-                }
-                IpAddressRef::IpV6AddressRef(_, ref ip_address_octets) => {
-                    untrusted::Input::from(ip_address_octets)
-                }
-            };
-            iterate_names(
-                // IP addresses are not compared against the subject field;
-                // only against Subject Alternative Names.
-                None,
-                cert.inner().subject_alt_name,
-                Err(Error::CertNotValidForName),
-                &|name| {
-                    #[allow(clippy::single_match)]
-                    match name {
-                        GeneralName::IpAddress(presented_id) => {
-                            match ip_address::presented_id_matches_reference_id(
-                                presented_id,
-                                ip_address,
-                            ) {
-                                Ok(true) => return NameIteration::Stop(Ok(())),
-                                Ok(false) => (),
-                                Err(_) => {
-                                    return NameIteration::Stop(Err(Error::BadDER));
-                                }
-                            }
+    let ip_address = match dns_name_or_ip {
+        DnsNameOrIpRef::DnsName(dns_name) => return verify_cert_dns_name(cert, dns_name),
+        DnsNameOrIpRef::IpAddress(IpAddrRef::V4(_, ref ip_address_octets)) => {
+            untrusted::Input::from(ip_address_octets)
+        }
+        DnsNameOrIpRef::IpAddress(IpAddrRef::V6(_, ref ip_address_octets)) => {
+            untrusted::Input::from(ip_address_octets)
+        }
+    };
+
+    iterate_names(
+        // IP addresses are not compared against the subject field;
+        // only against Subject Alternative Names.
+        None,
+        cert.inner().subject_alt_name,
+        Err(Error::CertNotValidForName),
+        &|name| {
+            #[allow(clippy::single_match)]
```

Comment:
Why not do `if let GeneralName::IpAddress(presented_id) = name` instead of the `match` with `#[allow(clippy::single_match)]`?

Reply:
Updated.